tests/test_transform.py

```diff
@@ -180,10 +296,126 @@ def checkTransform(self, fromName, toName):
                     transformClass(badPolyMap)
 
+    def checkTransformFromFrameSet(self, fromName, toName):
+        transformClassName = "Transform{}To{}".format(fromName, toName)
+        transformClass = getattr(afwGeom, transformClassName)
+        baseMsg = "transformClass={}".format(transformClass.__name__)
+        for nIn in self.goodNaxes[fromName]:
+            for nOut in self.goodNaxes[toName]:
+                msg = "{}, nIn={}, nOut={}".format(baseMsg, nIn, nOut)
+
+                baseFrame = makeGoodFrame(fromName, nIn)
+                currFrame = makeGoodFrame(toName, nOut)
+                frameSet = makeFrameSet(baseFrame, currFrame)
+                self.assertEqual(frameSet.getNframe(), 4)
+
+                # construct 0 or more frame sets that are invalid for this transform class
+                for badBaseFrame in makeBadFrames(fromName, nIn):
+                    badFrameSet = makeFrameSet(badBaseFrame, currFrame)
+                    with self.assertRaises(InvalidParameterError):
+                        transformClass(badFrameSet)
+                    for badCurrFrame in makeBadFrames(toName, nOut):
+                        reallyBadFrameSet = makeFrameSet(badBaseFrame, badCurrFrame)
+                        with self.assertRaises(InvalidParameterError):
+                            transformClass(reallyBadFrameSet)
+                for badCurrFrame in makeBadFrames(toName, nOut):
+                    badFrameSet = makeFrameSet(baseFrame, badCurrFrame)
+                    with self.assertRaises(InvalidParameterError):
+                        transformClass(badFrameSet)
+
+                transform = transformClass(frameSet)
+
+                desStr = "{}[{}->{}]".format(transformClassName, nIn, nOut)
+                self.assertEqual("{}".format(transform), desStr)
+                self.assertEqual(repr(transform), "lsst.afw.geom." + desStr)
+
+                frameSetCopy = transform.getFrameSet()
+
+                self.assertEqual(frameSet.getNframe(), frameSetCopy.getNframe())
+                self.assertEqual(frameSet.getFrame(1).getIdent(), "baseFrame")
+                self.assertEqual(frameSet.getFrame(2).getIdent(), "frame2")
+                self.assertEqual(frameSet.getFrame(3).getIdent(), "frame3")
+                self.assertEqual(frameSet.getFrame(4).getIdent(), "currFrame")
+
+                fromEndpoint = transform.getFromEndpoint()
+                toEndpoint = transform.getToEndpoint()
+                polyMap = makePolyMap(nIn, nOut)
+
+                # forward transformation of one point
+                rawInPoint = makeRawPointData(nIn)
+                inPoint = fromEndpoint.pointFromData(rawInPoint)
+                outPoint = transform.tranForward(inPoint)
+                rawOutPoint = toEndpoint.dataFromPoint(outPoint)
+                # TODO replace all use of np.allclose with assertFloatsAlmostEqual once DM-9707 is fixed
```

Comment:
Copy-and-pasted comment.

Can you factor out the common code here and in `checkTransformFromMapping` into a method that takes a `Transform`?